amp/amp-service-worker-offline-commenting.js

```diff
@@ -0,0 +1,49 @@
+/* global ERROR_MESSAGES, SITE_URL */
+{
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'amp-wpPendingComments' );
+	const errorMessages = ERROR_MESSAGES;
+
+	const commentHandler = ( { event } ) => {
+		const clonedRequest = event.request.clone();
+		return fetch( event.request )
+			.then( ( response ) => {
+				return response;
+			} )
+			.catch( () => {
+				return clonedRequest.blob().then( ( body ) => {
+					const queuedRequest = new Request( event.request.url, {
+						method: event.request.method,
+						headers: event.request.headers,
+						mode: event.request.mode,
+						credentials: event.request.credentials,
+						referrer: event.request.referrer,
+						redirect: 'follow',
+						body: body
+					} );
+
+					// Add request to queue. @todo Replace when upgrading to Workbox v4!
```

Comment:
This todo can now be _todone_ due to https://github.com/xwp/pwa-wp/pull/98

Reply:
Looks like we'll have to wait until [the related PR]( GoogleChrome/workbox#1710 ) gets merged, too.